File: Arcade/Intro/12_sortByHeight.js
function sortByHeight(a) {
    var h = a.slice(0);
    h.sort(function(a,b){return a-b});
    while (h[0] === -1) {
        h.shift();
    }
    
    if (h.length === a.length) { return h; }
    
    for (var i = 0; i < a.length; i++) {
        if (a[i] !== -1) {
            console.log(i);
            a[i] = h.shift();
        }
    } return a;
}

File: Arcade/Intro/C++/12_sortByHeight.cpp
std::vector<int> sortByHeight(std::vector<int> a) {
    vector<int> temp;
    for(int i:a) {
        if(i!=-1) temp.push_back(i);
    }
    sort(temp.begin(),temp.end());
    int j=0;
        for(int i=0;i<a.size();i++){
            if(a[i]!=-1) {
                a[i]=temp[j];
                j++;
                if(j>=temp.size()) break;
            }
        }
    return a;
}
